src/libraries/System.Threading.Channels/src/System/Threading/Channels/AsyncOperation.netstandard2.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Threading.Channels
+{
+    internal partial class AsyncOperation<TResult>
+    {
+        private void UnsafeQueueSetCompletionAndInvokeContinuation() =>
+            ThreadPool.UnsafeQueueUserWorkItem(static s => ((AsyncOperation<TResult>)s).SetCompletionAndInvokeContinuation(), this);
```

Comment:
This one makes sense.

Reply:
Yes. After all measurements, it is clear that this change is the only one that really optimizes memory and performance.